crates/ide/src/display.rs

```diff
@@ -41,7 +41,13 @@ pub(crate) fn function_declaration(node: &ast::Fn) -> String {
         format_to!(buf, "{}", type_params);
     }
     if let Some(param_list) = node.param_list() {
-        format_to!(buf, "{}", param_list);
+        let mut params = match param_list.self_param() {
+            Some(self_param) => vec![self_param.to_string()],
+            None => vec![],
+        };
+        params.extend(param_list.params().map(|param| param.to_string()));
```

Comment:
```suggestion
        let params = param_list.self_param().into_iter().chain(param_list.params().map(|param| param.to_string())).collect::<Vec<_>>();
```

`Option<T>` implements `IntoIterator`. This should be a bit faster as the `Vec` can be preallocated. You could also avoid the `.collect()` and use a `for` loop to append to `buf`. This would completely avoid the temporary `Vec`.

Reply:
Thanks for your suggestion. I changed to have the preallocated Vec. But keep the Vec to be able to use `join` method. I think it's better than rewritting the same logic than logic by checkin if we have the last element or not, etc...